File: HrmsWebApiCore/WebApiCore/Models/Bonus/FestivalBonusModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using WebApiCore.ViewModels.Bonus;

namespace WebApiCore.Models.Bonus
{
    public class FestivalBonusModel: FestivalBonusViewModel
    {
        public int? Grade { get; set; }
        public int? GradeID { get; set; }
        public decimal? OTPP { get; set; }
        public string Date { get; set; }
        public int? CompanyID { get; set; }
        public int? JobType { get; set;}
        public List<FestivalBonusViewModel> BonusGrid { get; set; }
       
    }
}

File: HrmsAngular/src/app/models/bonus/festival-bonus.model.ts
import { NgbDate, NgbDateStruct } from '@ng-bootstrap/ng-bootstrap';
import { BonusGridModel } from '../bonus-grid.model';
export class FestivalBonusModel{
  empCode:string;
  periodID:number;
  salaryHeadID:number;
  otpp:number;
  bonusType:number;
  date:any;
  dateNgb:NgbDateStruct;
  companyID:number;
  grade:number;
  depertment:string;
  depertmentID:number;
  designation:string;
  designationID:number;
  gradeID:number;
  jobType:number;
  companyid:number;
  bonusGrid:BonusGridModel[];

}
